Runtime/Game/Resources/LootLockerConfig.cs

```diff
@@ -1,37 +1,70 @@
 using System;
 using System.IO;
+using System.Linq;
 #if UNITY_EDITOR
+using System.Collections.Generic;
 using UnityEditor;
 using UnityEditor.PackageManager;
 using UnityEditor.PackageManager.Requests;
 using UnityEditor.PackageManager.UI;
+using LootLocker.Admin;
+using System.Text.RegularExpressions;
 #endif
 using UnityEngine;
 
 namespace LootLocker
 {
 
     public class LootLockerConfig : ScriptableObject
+
     {
 
         private static LootLockerConfig settingsInstance;
 
         public virtual string SettingName { get { return "LootLockerConfig"; } }
 
+#if UNITY_EDITOR
+        private void OnEnable()
+        {
+            ProjectSettings.APIKeyEnteredEvent += WriteConfigToDisk;
+            ProjectSettings.DomainKeyEnteredEvent += WriteConfigToDisk;
+        }
+
+        private void OnDisable()
+        {
+            ProjectSettings.APIKeyEnteredEvent -= WriteConfigToDisk;
+            ProjectSettings.DomainKeyEnteredEvent -= WriteConfigToDisk;
+        }
+#endif
```

Comment:
I think I need to be taught how all this works. We have more settings than api key and domain key. Game version for example is a game breaking setting that needs to be persisted.